daemon/api.go

```diff
@@ -2591,9 +2591,12 @@ func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
 	if err != nil {
 		return Forbidden("cannot get remote user: %s", err)
 	}
-	// we only allow "get" from regular users in snapctl
-	if uid != 0 && snapctlOptions.Args[0] != "get" {
-		return Forbidden("cannot use %q with uid %d, try with sudo", snapctlOptions.Args[0], uid)
+	// we only allow "get" and -h/--help from regular users in snapctl
+	if uid != 0 {
+		arg := snapctlOptions.Args[0]
```

Comment:
This is fine because we check for snapctlOptions.Args length above (expand diff context to see)

Reply:
Yeah, snapctl is a little peculiar, we have a very dumb client that doesn't do any arg processing and just passes all args to the server (this is completely different from snap command).
All the cli flags are handled server-side, they are passed down from here to hookstate/ctlcmd/